content/blogs/YYYY-MM-DD-CI-CD-kestra-comparison.md

```diff
@@ -0,0 +1,186 @@
+
+![Real time triggers in Kestra](/blogs/2024-06-27-realtime-triggers.jpg)
+
+- **Visual Workflow Editor**: While CI/CD tools often require you to dive into script configurations, Kestra offers a visual workflow editor. In fact, you can build and manage your workflows in a more intuitive, visual way, reducing the cognitive load of managing complex systems.
+
+![Workflows in Kestra](/blogs/YYYY-MM-DD-CI-CD-kestra-comparison/kestra_flows.jpg)
+
+## When to Use Kestra Over CI/CD Tools
+So, there’s no doubt that CI/CD tools like GitHub Actions, Jenkins, and Azure DevOps are powerful — especially when your needs are strictly related to CI/CD. But, as we've seen so far, Kestra comes into its own when you need to orchestrate more than just code deployments.
+
+So, here are some typical cases where you'd need to use Kestra rather than any CI/CD tool:
+
+- **Complex workflows**: When you have a workflow that has dependencies or requires coordination across different services, Kestra’s orchestration capabilities make it the better choice.
+
+- **Distributed systems**: For workflows that span multiple cloud environments or microservices, Kestra simplifies coordination and ensures tasks are completed in the right order, with built-in error handling.
+
+- **Error-prone workflows**: For workflows where errors and retries are expected, Kestra’s built-in error handling and retry mechanisms save you from writing custom scripts to manage failures.
+
+## Kestra Vs. Jenkins: A Deep Comparison
+So, along this article, we've described how Kestra differes from CI/CD tools and when you should choose it over them.
+
+Now it's time to get a practical example. To do so, we'll compare Kestra to Jenkins, and we'll:
+- Describe why developers use Jenkins.
+- Difficulties that developers encounter in using Jenkins and how Kestra can solve them.
+
+Let's start!
+
+### Why Developers Use Jenkins
+Jenkins has many pros, and we believe the three that are woth mentioning are:
+
+1. **Wide Adoption and Ecosystem**: Jenkins has been around for over a decade, so it is mature and "battle-tested": this makes it a staple in continuous integration and deployment. Also, its vast plugin ecosystem and open-source nature make it adaptable for almost any CI/CD pipeline need.
+
+2. **Flexibility**: With Jenkins, developers can create highly customized pipelines as it supports different environments, tools, and languages.
+
+3. **Community Support**: Jenkins has an active community that ensures there’s a large amount of documentation, tutorials, and forums where developers can find answers to their problems.
+
+Now, let's see where Jenkins falls short.
+
+### Difficulties Developers Encounter Using Jenkins
+While Jenkins is a powerful tool, it does come with some challenges.
+
+Let's see them together.
+
+#### Plugin Overload
+Jenkins currently has over 1,900 plugins, and this creates complexity as they can conflict with each other. Also, updating them can break pipelines, causing downtime.

```

Comment:
Same could be said for any tool with updates. What does Kestra do to help mitigate this?

How do they conflict with each other?